Week3/homework/step2-3.js

```diff
@@ -18,22 +19,28 @@ function repeatStringNumTimesWithWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  let i = 0;
+  while (i < num) {
+    result += str;
+    i++;
+  }
 
   return result;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
-
 // Use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  if (num > 0) {
```

Comment:
👍 for checking the value of `num` before entering the loop and skipping the look altogether is `num <= 0`.